shell/browser/electron_browser_main_parts.cc

```diff
@@ -320,7 +320,11 @@ int ElectronBrowserMainParts::PreCreateThreads() {
   }
 #endif
 
-  // Initialize the app locale.
+  // Fetch the system locale for Electron.
+  fake_browser_process_->SetSystemLocale(
```

Comment:
`l10n_util::GetApplicationLocale` will just return the preferred locale from the application bundle and not the actual system locale https://source.chromium.org/chromium/chromium/src/+/main:ui/base/l10n/l10n_util.cc;l=491 . Also for other platforms the system locale is only obtained via ICU when the API could find an application preferred locale https://source.chromium.org/chromium/chromium/src/+/main:ui/base/l10n/l10n_util.cc;l=528. I suggest the following changes instead,

1) Move `fake_browser_process_->SetSystemLocale` to the top of this function before `--lang` gets processed so that ICU default is not altered.

2) For windows and linux, call into `base::i18n::GetConfiguredLocale` which will obtain the value from `icu::Locale::getDefault` that does the necessary system calls ex: `GetLocaleInfoEx` refs, 
https://source.chromium.org/chromium/chromium/src/+/main:base/i18n/rtl.cc;l=96-99
https://source.chromium.org/chromium/chromium/src/+/main:third_party/icu/source/common/putil.cpp;l=1674-1951

3) For macOS, call into `[NSLocale currentLocale]`, refs https://developer.apple.com/documentation/foundation/nslocale/1409990-currentlocale . You might need to perform the `_` to `-` conversion still similar to https://source.chromium.org/chromium/chromium/src/+/main:ui/base/l10n/l10n_util_mac.mm;l=41-45 but please test if that is needed.

Reply:
I have confirmed that the `_` to `-` conversion is still needed by writing a small test app that prints the locale in XCode (it printed `en_GB`).